### ISSUE TITLE
JetBrains: Allow setting a default remote

### ISSUE THREAD

**Issue by vdavid** [sourcegraph/sourcegraph#34570]
## Problem

If there are multiple remotes for the current repo, the {right click on editor} → Sourcegraph → Copy link to file feature feature doesnn't know which repo to link to.

## Solution

@abeatrix wrote about the VS Code extension: "We actually have a setting `sourcegraph.defaultBranch` where you can set the default branch!"
That'd be perfect for the JetBrains plugin as well.

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.

**Comment by abeatrix** [comment]
@vdavid I have also implemented this setting for our Jetbrains plugin before if that helps 😄 https://github.com/sourcegraph/sourcegraph-jetbrains/commit/1fa8d5d6286c24924b55c15ed4d1a0b85ccab4d5

**Comment by vdavid** [comment]
@abeatrix, that's great, I didn't know that. Then what we should do in this issue is to move that setting to the Settings UI along with the server URL and access token fields.

**Comment by vdavid** [comment]
We actually have this, as Beatrix said:

![CleanShot 2022-06-02 at 18 21 13@2x](https://user-images.githubusercontent.com/2552265/171677528-1c94e7b0-7d81-4afe-a8e4-0859ccc5bb82.png)

So this issue is more about making sure that we're opening a branch that actually exists. PR coming up in minutes..

**Comment by jjinnii** [comment]
@vdavid Could I get more context on the UX of this issue? Is this related to[ VS Code's issue ](https://github.com/sourcegraph/sourcegraph/pull/35519)that if a user is trying to navigate to a file in Sourcegraph when the branch isn't available on Sourcegraph yet, we default to main?

**Comment by vdavid** [comment]
Yes, it's very similar!
Actually, we already had this functionality, including the check to see if the current branch exists on the remote. So there was nothing to do about this, really.

The only thing I did is that I switched the priority between the default branch and the current branch.
Because in the JetBrains plugin, we don't have this yet:
<img width="436" alt="CleanShot 2022-06-03 at 11 42 50@2x" src="https://user-images.githubusercontent.com/2552265/171830921-69d89f93-8c6b-4436-9fea-f3965255bcdb.png">
Basically, the default behavior used to be "Go to main branch", now it's "Go to current branch" (unless the current branch doesn't exist on the remote).

I also mentioned in the PR (https://github.com/sourcegraph/sourcegraph/pull/36496) that I'm not sure if this change is right, just it seems like a better option for me. But actually, I really like the VS Code solution, so I'll add an issue to change the JetBrains menu to this format.
